### PR TITLE
eskip: refactor parsing tests to check error message

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Check value of error message to simplify followup fixes.

E.g. #1885 introduced last route id to the error message without tests but last route could be empty or should not be present at all for `ParsePredicates` and `ParseFilters`.

### CODE REVIEW

File: eskip/eskip_test.go [hunk 1]
```diff
@@ -333,47 +210,62 @@ func TestParseFilters(t *testing.T) {
...
 		" \t",
 		nil,
-		false,
+		"",
 	}, {
 		"error",
 		"trallala",
 		nil,
-		true,
+		// TODO: remove empty last route id and fix position
```
Comment (AlexanderYastrebov) [inline eskip/eskip_test.go:223]: Position fix could be done by https://github.com/zalando/skipper/pull/2873

File: eskip/eskip_test.go [hunk 2]
```diff
@@ -7,297 +7,174 @@ import (
...
-}
-
-func TestParseRouteExpression(t *testing.T) {
+func TestParse(t *testing.T) {
 	for _, ti := range []struct {
 		msg        string
 		expression string
-		check      *Route
-		err        bool
+		check      []*Route
```
Comment (MustafaSaber) [inline eskip/eskip_test.go:17]: minor: maybe expectedRoutes would be better naming?

**Comment by MustafaSaber** [comment]
👍

File: eskip/eskip_test.go (hunk 2 above)
Comment (AlexanderYastrebov) [reply eskip/eskip_test.go:17]: I did not rename struct fields to keep diff small. All three test use different naming

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**